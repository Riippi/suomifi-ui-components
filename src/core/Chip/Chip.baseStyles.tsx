import { css } from 'styled-components';
import { withSuomifiTheme, TokensAndTheme, SuomifiThemeProp } from '../theme';
import { element, font, focus } from '../theme/reset';
import { disabledCursor } from '../../components/utils/css';

const removableStyles = ({ theme }: SuomifiThemeProp) => css`
  &.fi-chip--removable {
    & .fi-chip--content {
      max-width: 248px;
      margin-right: ${theme.spacing.xs};
    }
  }
`;

const disabledStyles = ({ theme }: SuomifiThemeProp) => css`
  &.fi-chip--disabled {
    &.fi-chip {
      ${disabledCursor}
      background: ${theme.colors.depthBase};
      &:hover,
      &:active {
        background: ${theme.colors.depthBase};
      }
    }
    & .fi-chip--icon {
      ${disabledCursor}
    }
  }
`;

export const baseStyles = withSuomifiTheme(
  ({ theme }: TokensAndTheme) => css`
    ${element({ theme })}
    ${font({ theme })('actionElementInnerTextBold')}
    ${focus({ theme })}

    &:focus{
      &:after {
      border-radius: 16px;
      }
    }

    &.fi-chip {
      border-radius: 14px;
      padding: ${theme.spacing.insetXxs} ${theme.spacing.insetL};
      color: ${theme.colors.whiteBase};
      background: ${theme.colors.highlightBase};
      max-height: 28px;
      display: inline-block;
    
      &.fi-chip--button{
        cursor: pointer;
        &:hover {
        background: ${theme.colors.highlightLight1};
        }

      &:active {
        background: ${theme.colors.highlightDark1};
        }
      }

    & .fi-chip--icon{
      height: 12px;
      width: 12px;
      transform: translateY(-0.35em);
    }

    & .fi-chip--content{
      display: inline-block;
      max-width: 270px;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      line-height: 1.5em;
      vertical-align: center;
    }

    }
    ${removableStyles({ theme })};
    ${disabledStyles({ theme })};
  `,
);
