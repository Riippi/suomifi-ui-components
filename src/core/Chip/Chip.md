```js
import { Chip } from 'suomifi-ui-components';

const removeAction = () => {
  alert('Selection removed');
};

<>
  <div styles={{ display: 'inline' }}>
    <Chip
      removable
      actionLabel="Unselect"
      onClick={removeAction}
      disabled
    >
      Selected item 1
    </Chip>

    <Chip removable actionLabel="Unselect" onClick={removeAction}>
      Selected item 2
    </Chip>

    <Chip onClick={removeAction}>Selected item 3</Chip>
  </div>

  <Chip.static>
    Selected item with a long content that exceeds the size limit
  </Chip.static>

  <Chip.static disabled>Disabled static chip</Chip.static>
</>;
```
