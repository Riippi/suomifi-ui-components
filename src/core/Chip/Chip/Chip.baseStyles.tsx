import { css } from 'styled-components';
import { defaultThemeTokens as theme } from '../../theme';
import { staticChipBaseStyles } from '../StaticChip.baseStyles';

export const baseStyles = css`
  ${staticChipBaseStyles}
  &.fi-chip--button {
    cursor: pointer;
    &:hover {
      background: ${theme.colors.highlightLight1};
    }

    &:active {
      background: ${theme.colors.highlightDark1};
    }
  }

  &.fi-chip--removable {
    & .fi-chip--content {
      max-width: 248px;
      margin-right: ${theme.spacing.xs};
    }
  }
`;
