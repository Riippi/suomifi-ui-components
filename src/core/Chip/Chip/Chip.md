```js
import { Chip } from 'suomifi-ui-components';
import React from 'react';

const removeAction = () => {
  alert('Selection removed');
};

const exampleRef = React.createRef();
const chipStyle = {
  display: 'inline-block',
  marginRight: '10px',
  marginBottom: '24px'
};
<>
  <div>
    <Chip
      style={chipStyle}
      removable
      actionLabel="Deselect item"
      onClick={removeAction}
    >
      Removable chip
    </Chip>
    <Chip
      style={chipStyle}
      removable
      actionLabel="Log referenced element"
      ref={exampleRef}
      onClick={() => console.log(exampleRef.current)}
    >
      Chip with ref
    </Chip>
  </div>
  <div>
    <Chip onClick={removeAction}>
      Clickable chip without remove icon
    </Chip>
  </div>

  <Chip>
    Chip with a long content that doesn't fit into the component's
    maximum width of 290px
  </Chip>

  <Chip disabled>Disabled chip</Chip>
</>;
```
