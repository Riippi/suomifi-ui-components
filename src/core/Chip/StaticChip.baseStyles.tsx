import { css } from 'styled-components';
import { defaultThemeTokens as theme } from '../theme';
import { element, font } from '../theme/reset';
import { absoluteFocus } from '../theme/utils/focus';

export const staticChipBaseStyles = css`
  ${element(theme)}
  ${font(theme)('actionElementInnerTextBold')}

    &:focus {
    outline: 0;
    position: relative;

    &::after {
      ${absoluteFocus}
      border-radius: 16px;
    }
  }

  &.fi-chip {
    border-radius: 14px;
    padding: ${theme.spacing.insetXxs} ${theme.spacing.insetL};
    color: ${theme.colors.whiteBase};
    background: ${theme.colors.highlightBase};
    max-height: 28px;
    display: inline-block;

    &.fi-chip--button {
      cursor: pointer;
      &:hover {
        background: ${theme.colors.highlightLight1};
      }

      &:active {
        background: ${theme.colors.highlightDark1};
      }
    }

    & .fi-chip--icon {
      height: 12px;
      width: 12px;
      transform: translateY(-0.35em);
    }

    & .fi-chip--content {
      display: inline-block;
      max-width: 270px;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      line-height: 1.5em;
      vertical-align: center;
    }

    &.fi-chip--disabled {
      &.fi-chip {
        cursor: not-allowed;
        background: ${theme.colors.depthBase};
        &:hover,
        &:active {
          background: ${theme.colors.depthBase};
        }
      }
      & .fi-chip--icon {
        cursor: not-allowed;
      }
    }
  }
`;
