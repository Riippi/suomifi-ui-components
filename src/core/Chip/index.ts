export { StaticChip, StaticChipProps } from './StaticChip/StaticChip';
export { Chip, ChipProps } from './Chip/Chip';
export { ChipList, ChipListProps } from './ChipList/ChipList';
