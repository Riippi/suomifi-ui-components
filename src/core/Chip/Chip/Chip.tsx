import React, { Component, forwardRef } from 'react';
import classnames from 'classnames';
import { default as styled } from 'styled-components';
import { HtmlButton, HtmlButtonProps, HtmlSpan } from '../../../reset';
import { VisuallyHidden } from '../../../components/Visually-hidden/Visually-hidden';
import { Icon } from '../../Icon/Icon';
import { logger } from '../../../utils/logger';
import { baseStyles } from './Chip.baseStyles';
import {
  BaseChipProps,
  baseClassName,
  chipClassNames,
} from '../BaseChip/BaseChip';

interface InternalChipProps
  extends BaseChipProps,
    Omit<
      HtmlButtonProps,
      'forwardedRef' | 'disabled' | 'onClick' | 'children'
    > {
  /**
   * Event handler to execute when clicked
   */
  onClick?: () => void;
  /**
   * Show X-icon next to the content to mark the chip as removable
   * @default false
   */
  removable?: boolean;
  /** Aria-label attribute to let screen reader users know pressing the button will remove the chip/selection  */
  actionLabel?: string;
}

interface InnerRef {
  forwardedRef: React.RefObject<HTMLButtonElement>;
}

export interface ChipProps extends InternalChipProps {
  /** Ref object to be passed to the button element */
  ref?: React.RefObject<HTMLButtonElement>;
}

class DefaultChip extends Component<ChipProps & InnerRef> {
  render() {
    const {
      className,
      children,
      onClick,
      removable,
      actionLabel,
      forwardedRef,
      disabled = false,
      ...passProps
    } = this.props;

    if (removable && !actionLabel) {
      logger.error(
        'Provide actionLabel to communicate removability to screen readers',
      );
    }
    return (
      <HtmlButton
        className={classnames(baseClassName, chipClassNames.button, className, {
          [chipClassNames.disabled]: !!disabled,
          [chipClassNames.removable]: !!removable,
        })}
        disabled={disabled}
        onClick={onClick}
        forwardedRef={forwardedRef}
        {...passProps}
      >
        <HtmlSpan className={chipClassNames.content}>{children}</HtmlSpan>
        {!!removable && (
          <>
            <Icon
              mousePointer={true}
              icon="close"
              color="currentColor"
              className={chipClassNames.icon}
              aria-hidden={true}
            />
            <VisuallyHidden>{actionLabel}</VisuallyHidden>
          </>
        )}
      </HtmlButton>
    );
  }
}

const StyledChip = styled(({ ...passProps }: InternalChipProps & InnerRef) => (
  <DefaultChip {...passProps} />
))`
  ${baseStyles}
`;

export const Chip = forwardRef(
  (props: ChipProps, ref: React.RefObject<HTMLButtonElement>) => {
    const { ...passProps } = props;
    return <StyledChip forwardedRef={ref} {...passProps} />;
  },
);
