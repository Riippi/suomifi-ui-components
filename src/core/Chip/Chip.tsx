import React, { Component, ReactNode } from 'react';
import classnames from 'classnames';
import { default as styled } from 'styled-components';
import { baseStyles } from './Chip.baseStyles';
import { HtmlButton, HtmlSpan } from '../../reset';
import { TokensProp, InternalTokensProp } from 'core/theme';
import { withSuomifiDefaultProps } from '../theme/utils';
import { Icon } from '../Icon/Icon';
import { logger } from '../../utils/logger';
import { VisuallyHidden } from '../../components/Visually-hidden/Visually-hidden';

const baseClassName = 'fi-chip';
const chipClassNames = {
  disabled: `${baseClassName}--disabled`,
  icon: `${baseClassName}--icon`,
  content: `${baseClassName}--content`,
  removable: `${baseClassName}--removable`,
  button: `${baseClassName}--button`,
};

type ChipVariant = 'static' | 'default';

interface InternalChipProps {
  /** Chip element content */
  children: ReactNode;
  /** Custom class name for styling and customizing  */
  className?: string;
  /** Disable chip */
  disabled?: boolean;
  /**
   * Event handler to execute when clicked
   */
  onClick?: () => void;
  /**
   * Show X-icon next to the content to mark the chip as removable
   * @default false
   */
  removable?: boolean;
  /** Aria-label attribute to let screen reader users know pressing the button will remove the chip/selection  */
  actionLabel?: string;
  /**
   * default | static - use default for an interactive chip and static for a purely visual chip.
   * @default default
   */
  variant?: ChipVariant;
}

export interface ChipProps extends InternalChipProps, TokensProp {}

class DefaultChip extends Component<ChipProps> {
  render() {
    const {
      className,
      children,
      onClick,
      removable,
      actionLabel,
      variant,
      disabled = false,
      ...passProps
    } = this.props;

    if (removable && !actionLabel) {
      logger.error(
        'Provide actionLabel to communicate removability to screen readers',
      );
    }

    if (variant === 'static') {
      return (
        <HtmlSpan
          className={classnames(baseClassName, className, {
            [chipClassNames.disabled]: !!disabled,
            [chipClassNames.removable]: !!removable,
          })}
          {...passProps}
        >
          <HtmlSpan className={chipClassNames.content}>{children}</HtmlSpan>
        </HtmlSpan>
      );
    }
    return (
      <HtmlButton
        className={classnames(baseClassName, chipClassNames.button, className, {
          [chipClassNames.disabled]: !!disabled,
          [chipClassNames.removable]: !!removable,
        })}
        disabled={disabled}
        onClick={onClick}
        {...passProps}
      >
        <HtmlSpan className={chipClassNames.content}>{children}</HtmlSpan>
        {!!removable && (
          <>
            <Icon
              mousePointer={true}
              icon="close"
              color="currentColor"
              className={chipClassNames.icon}
              aria-hidden={true}
            />
            <VisuallyHidden>{actionLabel}</VisuallyHidden>
          </>
        )}
      </HtmlButton>
    );
  }
}

const StyledChip = styled(
  ({ tokens, ...passProps }: ChipProps & InternalTokensProp) => (
    <DefaultChip {...passProps} />
  ),
)`
  ${(tokens) => baseStyles(withSuomifiDefaultProps(tokens))}
`;

export class Chip extends Component<ChipProps> {
  static static = (props: ChipProps) => {
    const passProps = withSuomifiDefaultProps(props);
    return <StyledChip {...passProps} variant="static" />;
  };

  render() {
    const { ...passProps } = withSuomifiDefaultProps(this.props);
    return <StyledChip {...passProps} />;
  }
}
