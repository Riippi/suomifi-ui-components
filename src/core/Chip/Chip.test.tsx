import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Chip as OrigChip, ChipProps } from './Chip';

const Chip = (props: ChipProps) => {
  const { children, ...passProps } = props;
  return <OrigChip {...passProps}>{children || 'Chip content'}</OrigChip>;
};

describe('disabled', () => {
  const DisabledChip = (
    <Chip disabled={true} onClick={() => null} actionLabel="Unselect">
      Testcontent
    </Chip>
  );

  it('has "--disabled"-class', () => {
    const { container } = render(DisabledChip);
    expect(container.firstChild).toHaveClass('fi-chip--disabled');
  });

  it('will add "disabled"-attribute for the button element', () => {
    const { getByRole } = render(DisabledChip);
    expect(getByRole('button')).toHaveAttribute('disabled');
  });

  it('should match snapshot', () => {
    const { container } = render(DisabledChip);
    expect(container).toMatchSnapshot();
  });
});

describe('children', () => {
  const chipWithChildren = (
    <Chip>
      <div data-testid="test-div">Selection 1</div>
    </Chip>
  );

  it('has the given content', () => {
    const { getByTestId } = render(chipWithChildren);
    expect(getByTestId('test-div').textContent).toBe('Selection 1');
  });

  it('should match snapshot', () => {
    const { container } = render(chipWithChildren);
    expect(container).toMatchSnapshot();
  });
});

describe('classnames', () => {
  const customClassChip = <Chip className="custom-class">Testcontent</Chip>;

  it('contains base classname', () => {
    const { container } = render(customClassChip);
    expect(container.firstChild).toHaveClass('fi-chip');
  });

  it('contains custom classname', () => {
    const { container } = render(customClassChip);
    expect(container.firstChild).toHaveClass('custom-class');
  });

  it('should match snapshot', () => {
    const { container } = render(customClassChip);
    expect(container).toMatchSnapshot();
  });
});

describe('variants', () => {
  const defaultChip = (
    <Chip removable={true} onClick={() => null} actionLabel="Unselect">
      Test content
    </Chip>
  );

  const staticChip = <OrigChip.static>Test content</OrigChip.static>;

  it('has icon element when set as removable', () => {
    const { container } = render(defaultChip);
    const icon = container.querySelector('.fi-chip--icon');
    expect(container.contains(icon)).toBeTruthy();
  });

  it('has the elements corresponding the chosen variant', () => {
    const { container } = render(staticChip);
    expect(container.firstChild).not.toHaveClass('fi-chip--button');
  });
});

describe('onClick', () => {
  it('is called when clicked', () => {
    const mockClick = jest.fn();
    const { getByRole } = render(
      <Chip onClick={mockClick} actionLabel="Unselect">
        Testcontent
      </Chip>,
    );
    const chip = getByRole('button');
    fireEvent.click(chip);
    expect(mockClick).toHaveBeenCalledTimes(1);
  });
});
