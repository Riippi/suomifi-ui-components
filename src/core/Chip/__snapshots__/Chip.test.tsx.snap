// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`children should match snapshot 1`] = `
.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c0:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c0::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c0::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c0::-webkit-inner-spin-button {
  height: auto;
}

.c0::-webkit-outer-spin-button {
  height: auto;
}

.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c1 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c1:focus {
  outline: 0;
  position: relative;
}

.c1:focus:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,55%);
  z-index: 9999;
}

.c1:focus:not(:focus-visible):after {
  content: none;
}

.c1:focus:after {
  border-radius: 16px;
}

.c1.fi-chip {
  border-radius: 14px;
  padding: 2px 10px;
  color: hsl(0,0%,100%);
  background: hsl(212,63%,45%);
  font: font-family:'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  font-weight: 600;
  max-height: 28px;
  display: inline-block;
  cursor: default;
}

.c1.fi-chip.fi-chip--button {
  cursor: pointer;
}

.c1.fi-chip.fi-chip--button:hover {
  background: hsl(212,63%,49%);
}

.c1.fi-chip.fi-chip--button:active {
  background: hsl(212,63%,37%);
}

.c1.fi-chip .fi-chip--icon {
  height: 12px;
  width: 12px;
  -webkit-transform: translateY(-0.35em);
  -ms-transform: translateY(-0.35em);
  transform: translateY(-0.35em);
}

.c1.fi-chip .fi-chip--content {
  display: inline-block;
  max-width: 270px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  line-height: 1.5em;
  vertical-align: center;
}

.c1.fi-chip--removable .fi-chip--content {
  max-width: 248px;
  margin-right: 10px;
}

.c1.fi-chip--disabled.fi-chip {
  background: hsl(202,7%,80%);
  cursor: not-allowed;
}

.c1.fi-chip--disabled.fi-chip:hover,
.c1.fi-chip--disabled.fi-chip:active {
  background: hsl(202,7%,80%);
}

<div>
  <button
    class="c0 fi-chip fi-chip--button c1"
    type="button"
  >
    <span
      class="c2 fi-chip--content"
    >
      <div
        data-testid="test-div"
      >
        Selection 1
      </div>
    </span>
  </button>
</div>
`;

exports[`classnames should match snapshot 1`] = `
.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c0:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c0::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c0::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c0::-webkit-inner-spin-button {
  height: auto;
}

.c0::-webkit-outer-spin-button {
  height: auto;
}

.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c1 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c1:focus {
  outline: 0;
  position: relative;
}

.c1:focus:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,55%);
  z-index: 9999;
}

.c1:focus:not(:focus-visible):after {
  content: none;
}

.c1:focus:after {
  border-radius: 16px;
}

.c1.fi-chip {
  border-radius: 14px;
  padding: 2px 10px;
  color: hsl(0,0%,100%);
  background: hsl(212,63%,45%);
  font: font-family:'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  font-weight: 600;
  max-height: 28px;
  display: inline-block;
  cursor: default;
}

.c1.fi-chip.fi-chip--button {
  cursor: pointer;
}

.c1.fi-chip.fi-chip--button:hover {
  background: hsl(212,63%,49%);
}

.c1.fi-chip.fi-chip--button:active {
  background: hsl(212,63%,37%);
}

.c1.fi-chip .fi-chip--icon {
  height: 12px;
  width: 12px;
  -webkit-transform: translateY(-0.35em);
  -ms-transform: translateY(-0.35em);
  transform: translateY(-0.35em);
}

.c1.fi-chip .fi-chip--content {
  display: inline-block;
  max-width: 270px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  line-height: 1.5em;
  vertical-align: center;
}

.c1.fi-chip--removable .fi-chip--content {
  max-width: 248px;
  margin-right: 10px;
}

.c1.fi-chip--disabled.fi-chip {
  background: hsl(202,7%,80%);
  cursor: not-allowed;
}

.c1.fi-chip--disabled.fi-chip:hover,
.c1.fi-chip--disabled.fi-chip:active {
  background: hsl(202,7%,80%);
}

<div>
  <button
    class="c0 fi-chip fi-chip--button c1 custom-class"
    type="button"
  >
    <span
      class="c2 fi-chip--content"
    >
      Testcontent
    </span>
  </button>
</div>
`;

exports[`disabled should match snapshot 1`] = `
.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c0:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c0::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c0::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c0::-webkit-inner-spin-button {
  height: auto;
}

.c0::-webkit-outer-spin-button {
  height: auto;
}

.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c1 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c1:focus {
  outline: 0;
  position: relative;
}

.c1:focus:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,55%);
  z-index: 9999;
}

.c1:focus:not(:focus-visible):after {
  content: none;
}

.c1:focus:after {
  border-radius: 16px;
}

.c1.fi-chip {
  border-radius: 14px;
  padding: 2px 10px;
  color: hsl(0,0%,100%);
  background: hsl(212,63%,45%);
  font: font-family:'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  font-weight: 600;
  max-height: 28px;
  display: inline-block;
  cursor: default;
}

.c1.fi-chip.fi-chip--button {
  cursor: pointer;
}

.c1.fi-chip.fi-chip--button:hover {
  background: hsl(212,63%,49%);
}

.c1.fi-chip.fi-chip--button:active {
  background: hsl(212,63%,37%);
}

.c1.fi-chip .fi-chip--icon {
  height: 12px;
  width: 12px;
  -webkit-transform: translateY(-0.35em);
  -ms-transform: translateY(-0.35em);
  transform: translateY(-0.35em);
}

.c1.fi-chip .fi-chip--content {
  display: inline-block;
  max-width: 270px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  line-height: 1.5em;
  vertical-align: center;
}

.c1.fi-chip--removable .fi-chip--content {
  max-width: 248px;
  margin-right: 10px;
}

.c1.fi-chip--disabled.fi-chip {
  background: hsl(202,7%,80%);
  cursor: not-allowed;
}

.c1.fi-chip--disabled.fi-chip:hover,
.c1.fi-chip--disabled.fi-chip:active {
  background: hsl(202,7%,80%);
}

<div>
  <button
    class="c0 fi-chip fi-chip--button c1 fi-chip--disabled"
    disabled=""
    type="button"
  >
    <span
      class="c2 fi-chip--content"
    >
      Testcontent
    </span>
  </button>
</div>
`;
